plugins/modules/volume.py

```diff
@@ -176,6 +176,33 @@ def create(self, resource):
         self._module.fail_on_missing_params(['name', 'size_gb'])
         return super(AnsibleCloudscaleVolume, self).create(resource)
 
+    def find_difference(self, key, resource, param):
+        is_different = False
+
+        if key == 'servers':
+            server_has = resource[key]
+            server_wanted = param
+            if len(param) > len(server_has):
+                is_different = True
+            else:
+                for has in server_has:
+                    if has["uuid"] not in param:
+                        is_different = True
+
+        # If it looks like a stub
+        elif isinstance(resource[key], dict) and 'href' in resource[key]:
```

Comment:
This is the same logic already implemented in the base module. You should defer to the base class using `super()`. If you want you could even check if the base class already determined a difference. If so, you don't need to check.

If there is no difference and the key is 'server', then you could do your check.